src/pubsub.c

```diff
@@ -223,6 +223,24 @@ int clientTotalPubSubSubscriptionCount(client *c) {
     return clientSubscriptionsCount(c) + clientShardSubscriptionsCount(c);
 }
 
+/* Returns 1 if a pattern is a prefix (at least 1 character with a single star at the end), 0 otherwise */
+int isPatternPrefix(robj *pattern) {
+    size_t pattern_len = sdslen(pattern->ptr);
+    return pattern_len > 1 && stringendswith(pattern->ptr,pattern_len,'*');
+}
+
+/* Add a star at the end of a channel prefix and return as a new robj.
+ * Caller is responsible for decrementing the reference count. */
+robj *patternFromPrefix(unsigned char *key, size_t key_len) {
+    size_t pattern_len = key_len + 1;
+    unsigned char *pattern_str = zmalloc(pattern_len);
```

Comment:
I got curious why the result of `zmalloc()` is never checked in the code and sure enough they terminate the process when it's out of memory.

Reply:
patternFromPrefix is removed